docs/src/content/docs/introduction/hash-verification.mdx

```diff
@@ -0,0 +1,45 @@
+---
+title: Hash verification
+---
+
+import { CodeBlock, LinkOut } from '@interledger/docs-design-system'
+
+Once a resource owner (RO) authorizes a client software, the authorization server (AS) will redirect the RO's [identity provider (IdP)](/introduction/idp/) to the finish URI. In order to secure this communication and verify that the redirect indeed emanated from the AS, the AS will provide a hash parameter in the request to the client's callback URI. The client **_must_** verify this hash.
+
+## Hashing method
+
+The hash base is generated by concatentating the following values in sequence using a single newline (0x0A) character to separate them:
+
+1. `nonce` value sent by the client in the initial request.
+2. `nonce` value sent by the AS from the interaction finish response.
```

Comment:
The interact nonce is returned after making the initial grant request (interaction instructions):

```json
{
  "interact": {
    "redirect": "https://auth.rafiki.money/4CF492MLVMSW9MKMXKHQ",
    "finish": "4105340a-05eb-4290-8739-f9e2b463bfa7" // <- nonce
  },
  "continue": {
    "access_token": {
      "value": "33OMUKMKSKU80UPRY5NM"
    },
    "uri": "https://auth.rafiki.money/continue/4CF492MLVMSW9MKMXKHQ",
    "wait": 30
  }
}
```